### ISSUE TITLE
ForkTsCheckerWebpackPlugin Configuration

### ISSUE THREAD

**Issue by MarZab** [AnomalyInnovations/serverless-bundle#127]
Currently, `ForkTsCheckerWebpackPlugin` is enabled when TypeScript is enabled.

While building, the service is run for each build, breaking the build when enough endpoints are defined:

```
Starting type checking service...
Starting type checking service...
Starting type checking service...
(node:6910) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
```

I would suggest a configuration option to not run ForkTsCheckerWebpackPlugin on build or to somehow run it before building.

**Comment by jayair** [comment]
Oh that's interesting. Does this error out the build?

**Comment by MarZab** [comment]
Yes, but not before freezing up my computer :D

**Comment by ashutoshkmr** [comment]
I am also facing the same issue

**Comment by jayair** [comment]
@MarZab @ashutoshkmr Can you create a simple repo that I can test with?

**Comment by MarZab** [comment]
The issue shows up with `package.individually = true`.

With trivial examples it takes a few more endpoints than with larger codebase for the it to come up.

https://github.com/MarZab/serverless-bundle-typescript-timeout

```
% npm run sls package

Serverless: Bundling with Webpack...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
Starting type checking service...
(node:71290) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
^C
```

**Comment by jayair** [comment]
Thanks for the repo.

I looked into it. It seems like serverless-webpack has been trying to resolve this issue (https://github.com/serverless-heaven/serverless-webpack/issues/299#issuecomment-496190541) and they have a proposed fix (https://github.com/serverless-heaven/serverless-webpack/pull/517) in their upcoming `5.4.0` release.

It serializes the compile process. I still see the Node warning about the memory leak. But the functions seem to get packaged.

I created a branch for you to test with:

``` bash
$ npm install --save-dev AnomalyInnovations/serverless-bundle#issue-127-forktschecker
```

And enable the option in your `serverless.yml` using:

``` yml
custom:
  bundle:
    serializedCompile: true
```

I don't think the solution is ideal. I'd prefer being able to set the concurrency instead. But give it a shot.

**Comment by MarZab** [comment]
I tried on my project, and it did complete while also showing the memory leak. 

But it took a lot longer so I did some testing. added an option to disable forktschecker [here](https://github.com/MarZab/serverless-bundle/tree/issue-127-forktschecker).

Config | Build Time | System Load
-------|------------|-------------- 
tsForkTsChecker: false serializedCompile: false | 69.41s |  10
tsForkTsChecker: true serializedCompile: true | 375.33s | 21
tsForkTsChecker: true serializedCompile: false | frooze | 427
tsc --noEmit | 20.25s |  7

My solution for now is to not use serializedCompile and to disable tsForkTsChecker compile time.

**Comment by jayair** [comment]
Yeah the `serializedCompile` will take longer. This option isn't directly related to forktschecker but it's a problem that a lot of people that use the `package.individually` option have. From having a look at the serverless-webpack 5.4.0 release roadmap, it seems like an issue they are trying to tackle. I'm leaning towards waiting for them to take a crack at it.

Maybe for the time being hang on to your solution?

**Comment by Namstel** [comment]
We're also running into this issue, but so far the only issue we have is that it slows down deployment by a lot.

Deployment with tsForkTsChecker turned off is 19 seconds and deployment with it turned on is 49 seconds.

I'm looking forward to seeing @MarZab 's solution in serverless-bundle.

**Comment by jayair** [comment]
@Namstel So it's faster with ForkTsChecker turned off?

**Comment by Namstel** [comment]
@jayair Yes definitely.

**Comment by jayair** [comment]
We are hearing a ton of folks having memory related issues because of this plugin. Anybody want to take a crack at this and making ForkTsChecker an option instead?

**Comment by ashutoshkmr** [comment]
Hey @jayair 
I've created a [PR](https://github.com/AnomalyInnovations/serverless-bundle/pull/185) to make ForkTsChecker an option

**Comment by ceoshikhar** [comment]
> Hey @jayair
> I've created a [PR](https://github.com/AnomalyInnovations/serverless-bundle/pull/185) to make ForkTsChecker an option

@jayair Please have a look at the PR. I have been having a really bad time with this memory leak error. My deployments pipeline fail 9/10 times because it crashes the entire process and if I deploy it on my machine it freezes my computer. I hope this gets fixed soon.

**Comment by jayair** [comment]
Yup. I’ll be looking at this today. Thanks!

**Comment by jayair** [comment]
Hi all, we made this an option:

https://github.com/AnomalyInnovations/serverless-bundle/releases/tag/v4.2.0

Read more on the option here: https://github.com/AnomalyInnovations/serverless-bundle#forktscheckerwebpackplugin

Can you try it and let me know if it fixes your issues?

**Comment by ceoshikhar** [comment]
@jayair I am currently on version `3.0.0`, will it break anything if I upgrade to `4.2.0` ?
